resolwe/flow/tests/test_manager.py

```diff
@@ -96,6 +99,17 @@ def test_workflow(self):
             {DataDependency.KIND_SUBPROCESS},
         )
 
+    def test_communicate(self):
+        """Make sure communicate can serialize settings.
+
+        When object of PosixPath is used in settings the call to communicate
+        fails due to object not being JSON serializable by default.
+        """
+        with override_settings(FLOW_EXECUTOR={"DATA_DIR": Path("/some/path")}):
+            async_to_sync(manager.communicate)(run_sync=True)
+        # Make sure original settings are saved.
```

Comment:
Are settings not saved by default in tests? If not, that can cause a lot of problems.

Reply:
They are not saved by default when communicate is called from channels (like from listener, for example). So if original settings are not saved back (the last line), then all other tests fail.

Reply:
It would be great to force-update settings for each test. But that can be done in a separate pull request.